### ISSUE TITLE
Empty Cookie header causes errors

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#3965]
Issue reported in OWASP ZAP User Group: https://groups.google.com/d/topic/zaproxy-users/QsT-8DPb41w/discussion

Steps to reproduce the issue:
1. Run ZAP;
2. Send/proxy a HTTP request with an empty Cookie header;
3. Note several errors logged:
```
497034 [Thread-7] ERROR org.parosproxy.paros.network.HttpSender  - 
java.lang.IllegalArgumentException: 
	at org.parosproxy.paros.network.HtmlParameter.<init>(Unknown Source)
	at org.parosproxy.paros.network.HttpRequestHeader.getCookieParams(Unknown Source)
	at org.parosproxy.paros.network.HttpRequestHeader.getHttpCookies(Unknown Source)
	at org.zaproxy.zap.extension.httpsessions.ExtensionHttpSessions.onHttpRequestSend(Unknown Source)
	at org.parosproxy.paros.network.HttpSender.notifyRequestListeners(Unknown Source)
	at org.parosproxy.paros.network.HttpSender.sendAndReceive(Unknown Source)
	at org.parosproxy.paros.network.HttpSender.sendAndReceive(Unknown Source)
	at org.parosproxy.paros.extension.manualrequest.http.impl.HttpPanelSender.handleSendMessage(Unknown Source)
	at org.parosproxy.paros.extension.manualrequest.ManualRequestEditorDialog$3.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)
499823 [ZAP-PassiveScanner] ERROR org.zaproxy.zap.extension.params.ExtensionParams  - 
java.lang.IllegalArgumentException: 
	at org.parosproxy.paros.network.HtmlParameter.<init>(Unknown Source)
	at org.parosproxy.paros.network.HttpRequestHeader.getCookieParams(Unknown Source)
	at org.parosproxy.paros.network.HttpMessage.getCookieParams(Unknown Source)
	at org.zaproxy.zap.extension.params.ExtensionParams.onHttpRequestSend(Unknown Source)
	at org.zaproxy.zap.extension.params.ParamScanner.scanHttpRequestSend(Unknown Source)
	at org.zaproxy.zap.extension.pscan.PassiveScanThread.run(Unknown Source)
499823 [ZAP-PassiveScanner] ERROR org.zaproxy.zap.extension.pscan.PassiveScanThread  - Scanner Parameter Scanner failed on record 1 from History table: GET http://example.com/
java.lang.IllegalArgumentException: 
	at org.parosproxy.paros.network.HtmlParameter.<init>(Unknown Source)
	at org.parosproxy.paros.network.HttpRequestHeader.getCookieParams(Unknown Source)
	at org.parosproxy.paros.network.HttpMessage.getCookieParams(Unknown Source)
	at org.zaproxy.zap.extension.params.ExtensionParams.onHttpResponseReceive(Unknown Source)
	at org.zaproxy.zap.extension.params.ParamScanner.scanHttpResponseReceive(Unknown Source)
	at org.zaproxy.zap.extension.pscan.PassiveScanThread.run(Unknown Source)
```
ZAP should cope better with the malformed cookie header.

ZAP:
Version 2.6.0 (and older).

**Comment by AmitKulkarni9** [comment]
Thanks thc202 for fixing this. This will be available in ver 2.7.0 ? When is the release for it ?

Thanks,
Amit

**Comment by thc202** [comment]
Yes, the fix will be included in 2.7.0. Probably next week, if there's no issue with the release candidate.
